src/components/encounter/ohri-encounter-form.component.tsx

```diff
@@ -544,6 +553,42 @@ export const OHRIEncounterForm: React.FC<OHRIEncounterFormProps> = ({
         return Promise.all(saveAttachmentPromises).then(() => response);
       });
     }
+
+    if (encounterForSubmission.obs?.length || encounterForSubmission.orders?.length) {
+      if (Object.keys(encounterContext?.patientIdentifier).length) {
```

Comment:
A better approach would be traversing through all form fields and collect ID values for fields with the patient ID datatype.